### DISCUSSION TITLE
How to draw in `xtd`?

### DISCUSSION THREAD

**Discussion by ghost** [gammasoft71/xtd#238] [Q&A] [answered]
For example, I want to plot the sine wave. Please give me complete sample code. Thank you.

**Comment by gammasoft71** [comment, marked as answer]
The following example shows how to draw a sine wave on a form.

```c++
#include <xtd/drawing/pen>
#include <xtd/forms/application>

using namespace xtd;
using namespace xtd::drawing;
using namespace xtd::forms;

auto main()->int {
  const auto sins = std::vector<point> {{10, 110}, {20, 148}, {30, 181}, {40, 202}, {50, 210}, {60, 202}, {70, 181}, {80, 148}, {90, 110}, {100, 72}, {110, 39}, {120, 18}, {130, 10}, {140, 18}, {150, 39}, {160, 72}, {170, 110}, {180, 148}, {190, 181}, {200, 202}, {210, 210}, {220, 202}, {230, 181}, {240, 148}, {250, 110}, {260, 72}, {270, 39}, {280, 18}, {290, 10}, {300, 18}, {310, 39}, {320, 72}};
  auto form1 = form::create("Sine wave");
  form1.client_size({330, 220});
  form1.paint += [&](object& sender, paint_event_args& e) {
    e.graphics().draw_curve(pen(color::dodger_blue, 4), sins);
  };
  xtd::forms::application::run(form1);
}
```

* You can see [How to draw something on control?](https://github.com/gammasoft71/xtd/wiki/How-to-draw-something-on-control-%3F) which shows different ways of drawing with xtd.
* See [form_paint](https://github.com/gammasoft71/xtd/blob/master/examples/xtd.forms.examples/events/form_paint/README.md) for an example of how to use other methods in the [xtd::drawing::graphics](https://gammasoft71.github.io/xtd/reference_guides/latest/classxtd_1_1drawing_1_1graphics.html) class.

**Reply by ghost** [reply]
> * You can see [How to draw something on control?](How-to-draw-something-on-control-%3F) which shows different ways of drawing with xtd.

Page Not Found my friend.

**Reply by ghost** [reply]
> The following example shows how to draw a sine wave on a form.
> 
> ```c++
> #include <xtd/drawing/pen>
> #include <xtd/forms/application>
> 
> using namespace xtd;
> using namespace xtd::drawing;
> using namespace xtd::forms;
> 
> auto main()->int {
>   const auto sins = std::vector<point> {{10, 110}, {20, 148}, {30, 181}, {40, 202}, {50, 210}, {60, 202}, {70, 181}, {80, 148}, {90, 110}, {100, 72}, {110, 39}, {120, 18}, {130, 10}, {140, 18}, {150, 39}, {160, 72}, {170, 110}, {180, 148}, {190, 181}, {200, 202}, {210, 210}, {220, 202}, {230, 181}, {240, 148}, {250, 110}, {260, 72}, {270, 39}, {280, 18}, {290, 10}, {300, 18}, {310, 39}, {320, 72}};
>   auto form1 = form::create("Sine wave");
>   form1.client_size({330, 220});
>   form1.paint += [&](object& sender, paint_event_args& e) {
>     e.graphics().draw_curve(pen(color::dodger_blue, 4), sins);
>   };
>   xtd::forms::application::run(form1);
> }
> ```

This is using a fixed set of points. What about using the sine function itself?

**Reply by gammasoft71** [reply]
The [xtd::drawing::graphics](https://gammasoft71.github.io/xtd/reference_guides/latest/classxtd_1_1drawing_1_1graphics.html) class doesn't contain any sinusoidal functions per se. The [xtd::drawing::graphics::draw_curve](https://gammasoft71.github.io/xtd/reference_guides/latest/classxtd_1_1drawing_1_1graphics.html#a21d44c0172f4cb6e9a80c733e93e090f) method is used to draw a curve. Just as the [xtd::math::sin](https://gammasoft71.github.io/xtd/reference_guides/latest/classxtd_1_1math.html#a9a4285e5c81dd09375d7149c36401563) function is only used to calculate a sine. If you want a method that generates and draws, you've got all the tools you need :-).

*Just for information, xtd doesn't yet have charts.*

PS : Link fixed 
> You can see [How to draw something on control?](https://github.com/gammasoft71/xtd/wiki/How-to-draw-something-on-control-%3F) which shows different ways of drawing with xtd.
